dashboards-observability/public/components/visualizations/charts/data_table/data_table.tsx

```diff
@@ -3,102 +3,487 @@
-    const rowEnd = Math.min(rowStart + pagination.pageSize, data.length);
-    return data.slice(rowStart, rowEnd);
-  }, [data, pagination]);
+  const columnVisiblityHandler = useCallback((visible: boolean, feild: any) => {
+    const isExisting = columnVisibility.findIndex((i) => i === feild);
+    if (visible) {
+      if (isExisting > -1) {
+        columnVisibility.splice(isExisting, 1);
+        gridRef?.current?.columnApi?.setColumnsVisible([feild], true);
+      }
+    } else {
+      columnVisibility.push(feild);
+      gridRef?.current?.columnApi?.setColumnsVisible([feild], false);
+    }
+    setColumnVisibility([...columnVisibility]);
+  }, []);
 
-  const columns = fields.map((field) => {
-    return {
-      id: field.name,
+  // pagination
+  const pageCount = Math.ceil(raw_data.length / pageSize);
+  const [activePage, setActivePage] = useState(0);
+  const goToPage = ({ selected }: { selected: number }) => {
+    setActivePage(selected);
+    gridRef.current.api.paginationGoToPage(selected);
+    if (isFullScreen) {
+      gridRefFullScreen.current.api.paginationGoToPage(selected);
+    }
+  };
+
+  const setIsFullScreenHandler = (val: boolean) => {
+    setIsFullScreen(val);
+  };
+
+  useEffect(() => {
+    document.addEventListener('keydown', hideGridFullScreenHandler);
+    return () => {
+      document.removeEventListener('keydown', hideGridFullScreenHandler);
     };
-  });
+  }, []);
 
-  // Column visibility
-  const [visibleColumns, setVisibleColumns] = useState(columns.map(({ id }) => id));
+  const hideGridFullScreenHandler = (e: any) => {
+    if (e.key === 'Escape') {
+      setIsFullScreen(false);
+    }
+  };
+  return (
+    <>
+      <GridHeader
+        isFullScreen={isFullScreen}
+        setIsFullScreenHandler={setIsFullScreenHandler}
+        selectedRowDensity={selectedRowDensity}
+        selectDensityHandler={selectDensityHandler}
+        columnVisiblityHandler={columnVisiblityHandler}
+        columns={columns}
+        columnVisibility={columnVisibility}
+      />
+      <div style={{ height: '460px' }}>
+        <AgGridReact
+          ref={gridRef}
+          rowData={raw_data}
+          columnDefs={columns}
+          defaultColDef={defaultColDef}
+          animateRows={true}
+          rowSelection="multiple"
+          enableRangeSelection={true}
+          pagination={true}
+          paginationPageSize={pageSize}
+          paginationNumberFormatter={paginationNumberFormatter}
+          suppressPaginationPanel={true}
+          rowHeight={selectedRowDensity.height}
+        />
+      </div>
+      <GridFooter
+        onPageSizeChanged={onPageSizeChanged}
+        pageSize={pageSize}
+        activePage={activePage}
+        goToPage={goToPage}
+        pageCount={pageCount}
+      />
+      {isFullScreen && (
+        <CustomOverlay>
+          <EuiFlexGroup direction="column">
+            <EuiFlexItem>
+              <div style={{ height: '500px' }}>
```

Comment:
Here also you are hardcoding the height. This is not ideal. If you really want to do this, create some config variables and then use those variables here. What you can do is to create another issue to handle dynamic heights for tables, if this becomes difficult to achieve in a day.